packages/pigeon/lib/swift_generator.dart

```diff
@@ -608,9 +608,13 @@ import FlutterMacOS
         // Special-cased to avoid warnings about using 'as' with Any.
         return value;
       } else if (type.baseName == 'int') {
-        final String orString =
-            type.isNullable ? 'nilOrValue($value)' : '$value as! Int64';
-        return '($value is Int32) ? Int64($value as! Int32) : $orString';
+        if (type.isNullable) {
+          // Nullable ints need to check for NSNull, nil, and Int32 before casting can be done safely.
+          // This nested ternary is a necessary evil to avoid less efficient conversions.
+          return '$value is NSNull ? nil : ($value is Int64 ? $value as! Int64? : Int64($value as! Int32))';
```

Comment:
There's a typo here relative to my version: it needs to be `$value is Int64? ?` not `$value is Int64 ?`; as written this will crash on `nil`—our integration tests aren't catching it because we're currently always using `NSNull`.

(It would crash because `nil is NSNull` is false, `nil is Int64' is false, and so it would get all the way to `Int64(nil as! Int32))` and `nil as! Int32` is invalid.)

Reply:
oops

Reply:
This is clean! Alternatively you can also write a helper function with if-else statement (similar to `nilOrValue<T>` you have), if you wanna make it more readable.

Reply:
I was considering that as well. 

I was going to make a generic version, but handling multiple generic types in swift seems essentially impossible.

Reply:
> I was going to make a generic version, but handling multiple generic types in swift seems essentially impossible.

This doesn't have to be generic, since we are dealing with concrete types (`Int32/Int64`). 

Just for fun, if you wanna try generic function, since you can't return different types, you will have to use "out params" (`inout` keyword), which is actually harder to use in caller. 

But let's keep the current inline implementation and land this. It's look good enough.

Reply:
> This doesn't have to be generic, since we are dealing with concrete types (`Int32/Int64`).

I know, I was just hoping to add it to the nilOrValue function to avoid adding an Int only casting function. I think as is works best